### ISSUE TITLE
angular/di-unused: does not work with array syntax?

### ISSUE THREAD

**Issue by sbalay** [Gillespie59/eslint-plugin-angular#287]
**di-unused** rule does not seem to work when using the array syntax for dependency injection. Would it be possible to extend the rule to support array syntax?

**Comment by remcohaszing** [comment]
`eslint-plugin-angular` version `0.14.0` has been released about 2 hours ago. Try bumping to this version. If this doesn't help, please provide an example which doesn't work as expected.

**Comment by sbalay** [comment]
I've tested this with a file named *percentageFilter.js* which contains the following code
```
angular.module('wtravel-sales').filter('percentage', [
  function () {
    return value => value * 100;
  }
]);
```
In that case, no warning is displayed when running ```eslint percentageFilter.js```

After changing the content of the file to
```
angular.module('wtravel-sales').filter('percentage', [
  '$translate',
  function ($translate) {
    return value => value * 100;
  }
]);
```
Still, no warning is displayed, although, the $translate dependency is never used

**Comment by sbalay** [comment]
Forgot to mention that I'm using the version ```0.14.0``` of eslint-plugin-angular

**Comment by remcohaszing** [comment]
I am unable to reproduce this bug.

Since you reported this about an hour after the release which should have fixed this, it sounds very likely to me that you're running an older version.

* Is it possible that you bumped the version in `eslint-plugin-angular` in your `package.json`, but forgot to run `npm install`?
* Do you try to run a globally installed `eslint` while using a locally installed `eslint-plugin-angular`? This doesn't work. If so, try `node_modules/.bin/eslint percentageFilter.js`.

The two examples you provided both use the array syntax. One without any dependencies, the other with a dependency. Do the following variants get reported correctly?

*should get reported*:
```js
angular.module('wtravel-sales').filter('percentage', 
  function ($translate) {
    return value => value * 100;
  }
);
```

*should get reported*:
```js
var app = angular.module('wtravel-sales');
app.filter('percentage', 
  function ($translate) {
    return value => value * 100;
  }
);
```

*should not get reported*:
```js
var app = null;
app.filter('percentage', 
  function ($translate) {
    return value => value * 100;
  }
);
```

**Comment by sbalay** [comment]
@remcohaszing thanks for the quick responses.
Didn't understand why your last example shouldn't get reported.
```
var app = null;
app.filter('percentage', 
  function ($translate) {
    return value => value * 100;
  }
);
```
That code _should_ get reported because the $translate dependency is being injected but is not being used.

In my case, I have the following case which I think should get reported but it is not happening

```
angular.module('myapp').filter('myfilter', [
  '$translate', '$filter',
  function ($translate, $filter) {
    return function (value) {
      return $filter(value, 4) * 100;
    }
  }
]);
```
This should be invalid because the $translate dependency is being injected and yet it is never used.

**Comment by tilmanpotthof** [comment]
@sbalay I added your example as unit test (commit 050fb45). Unfortunately, the test does not fail. I copied your example to some real app and also got the correct error report: `Unused injected value $translate`

Based on these information I'm pretty sure your problem is caused by a configuration or installation problem.

* Try to remove your `node_modules` folder and rerun `npm install`
* How do you run the rules? ESLint itself? Grunt? Gulp?

If your run a global eslint installation, the plugin is also taken from the global node_modules and your package.json is ignore. You can check this with `npm list -g eslint-plugin-angular` and fix it with `npm upgrade -g eslint-plugin-angular`

I'll close this issue, but if you identified the problem please report it here. Probably we have to improve the documentation about version updates in different setups.
